src/Uno.Foundation/Uno.Core.Extensions/PooledObjects/PooledStringBuilder.cs

```diff
@@ -0,0 +1,104 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+// https://github.com/dotnet/roslyn/blob/cd5e6895d3dc8dfad270505b37d28be70077a645/src/Dependencies/PooledObjects/PooledStringBuilder.cs
+
+#nullable enable
+
+using System.Diagnostics;
+using System.Text;
+
+namespace Microsoft.CodeAnalysis.PooledObjects
+{
+    /// <summary>
+    /// The usage is:
+    ///        var inst = PooledStringBuilder.GetInstance();
+    ///        var sb = inst.builder;
+    ///        ... Do Stuff...
+    ///        ... sb.ToString() ...
+    ///        inst.Free();
+    /// </summary>
+    internal sealed partial class PooledStringBuilder
+    {
+        public readonly StringBuilder Builder = new();
+        private readonly ObjectPool<PooledStringBuilder> _pool;
+
+        private PooledStringBuilder(ObjectPool<PooledStringBuilder> pool)
+        {
+            Debug.Assert(pool != null);
+            _pool = pool!;
+        }
+
+        public int Length
+        {
+            get { return this.Builder.Length; }
+        }
+
+        public void Free()
+        {
+            var builder = this.Builder;
+
+            // do not store builders that are too large.
+            if (builder.Capacity <= 1024)
+            {
+                builder.Clear();
+                _pool.Free(this);
+            }
+            else
+            {
+                _pool.ForgetTrackedObject(this);
+            }
```

Comment:
This is the implementation that Roslyn has.

I guess the 1024 threshold is too small for us that we'll end up forgetting most of the builders, so we get only the pooling overhead without the actual benefit of reusing objects. I'll look into either storing all builders or make the threshold a little bit higher.

Reply:
Sure. Let's evaluate what the unlimited use is when building the samples app (which has a very large amount of XAML files).